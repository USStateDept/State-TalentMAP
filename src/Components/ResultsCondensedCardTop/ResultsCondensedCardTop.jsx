import React from 'react';
import { Link } from 'react-router-dom';
import FontAwesome from 'react-fontawesome';
import { POSITION_DETAILS, HOME_PAGE_CARD_TYPE } from '../../Constants/PropTypes';

const ResultsCondensedCardTop = ({ position, type }) => {
  let icon = '';
  let cardTopClass = '';
  let useType = false;
  if (type === 'serviceNeed') {
    icon = 'bolt';
    cardTopClass = 'card-top-alternate';
    useType = true;
  }
  return (
    <div className={`usa-grid-full condensed-card-top ${cardTopClass}`}>
      <div className="usa-grid-full condensed-card-top-header-container">
        <div
          className={
            'usa-width-two-thirds condensed-card-top-header condensed-card-top-header-left'
          }
        >
          {useType && <span><FontAwesome name={icon} /> | </span>}
          {position.title}
        </div>
        <div
          className="usa-width-one-third condensed-card-top-header condensed-card-top-header-right"
        >
          Grade: {position.grade}
        </div>
      </div>
      <div className="usa-grid-full">
        <Link to={`/details/${position.position_number}`}>View More</Link>
      </div>
    </div>
  );
};

ResultsCondensedCardTop.propTypes = {
  position: POSITION_DETAILS.isRequired,
  type: HOME_PAGE_CARD_TYPE.isRequired,
};

ResultsCondensedCardTop.defaultProps = {
  type: 'default',
};

export default ResultsCondensedCardTop;
