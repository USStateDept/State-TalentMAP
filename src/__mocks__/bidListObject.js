const bidListObject = {
  count: 3,
  next: null,
  previous: null,
  results: [
    {
      id: 3,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 226,
        grade: '03',
        skill: 'POLITICAL AND ECONOMIC AFFAIRS (6050)',
        position_number: '10035561',
        title: 'POLITICAL/ECONOMIC OFFICER',
        create_date: '2015-08-05',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 199,
          location: {
            id: 103,
            country: 'Sierra Leone',
            code: '00A',
            city: 'Freetown',
            state: '',
          },
        },
      },
      status: 'approved',
      draft_date: '2017-12-20',
      submitted_date: '2017-12-20',
      handshake_offered_date: '2017-12-20',
      handshake_accepted_date: '2017-12-20',
      handshake_declined_date: '2017-12-20',
      in_panel_date: '2017-12-20',
      scheduled_panel_date: '2017-12-20',
      approved_date: '2017-12-20',
      declined_date: '2017-12-20',
      closed_date: '2017-12-20',
      create_date: '2017-12-20',
      update_date: '2017-12-20',
      is_priority: false,
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 1,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 1,
        grade: '05',
        skill: 'OFFICE MANAGEMENT (9017)',
        position_number: '00003026',
        title: 'OMS (COM)',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 235,
          location: {
            id: 103,
            country: 'Sierra Leone',
            code: '00A',
            city: 'Freetown',
            state: '',
          },
        },
      },
      status: 'closed',
      draft_date: '2017-12-20',
      submitted_date: '2017-12-20',
      handshake_offered_date: '2017-12-20',
      handshake_accepted_date: '2017-12-20',
      handshake_declined_date: '2017-12-20',
      in_panel_date: '2017-12-20',
      scheduled_panel_date: '2017-12-20',
      approved_date: '2017-12-20',
      declined_date: '2017-12-20',
      closed_date: '2017-12-20',
      create_date: '2017-12-20',
      update_date: '2017-12-20',
      is_priority: true,
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
    {
      id: 2,
      bidcycle: 'Demo BidCycle 2017-10-31 16:00:51.254762',
      user: 'rehmant',
      position: {
        id: 84,
        grade: '04',
        skill: 'SECURITY (2501)',
        position_number: '56001600',
        title: 'SPECIAL AGENT',
        create_date: '2006-09-20',
        update_date: '2017-06-08',
        bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
        post: {
          id: 195,
          location: {
            id: 103,
            country: 'Sierra Leone',
            code: '00A',
            city: 'Freetown',
            state: '',
          },
        },
      },
      status: 'closed',
      draft_date: '2017-12-20',
      submitted_date: '2017-12-20',
      handshake_offered_date: '2017-12-20',
      handshake_accepted_date: '2017-12-20',
      handshake_declined_date: '2017-12-20',
      in_panel_date: '2017-12-20',
      scheduled_panel_date: '2017-12-20',
      approved_date: '2017-12-20',
      declined_date: '2017-12-20',
      closed_date: '2017-12-20',
      create_date: '2017-12-20',
      update_date: '2017-12-20',
      is_priority: false,
      reviewer: {
        username: 'woodwardw',
        first_name: 'Wendy',
        last_name: 'Woodward',
        email: 'woodwardw@state.gov',
        phone_number: '555-555-5555',
        is_cdo: false,
      },
    },
  ],
};

export default bidListObject;
