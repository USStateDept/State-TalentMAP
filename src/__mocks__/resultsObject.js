const resultsObject = {
  count: 2,
  results: [
    {
      id: 6,
      grade: '06',
      skill: 'OFFICE MANAGEMENT (9017)',
      bureau: '(AF) BUREAU OF AFRICAN AFFAIRS',
      organization: '(MONROVIA) MONROVIA LIBERIA',
      classifications: [],
      representation: '[10034001] OMS (DCM) (Monrovia, Liberia)',
      is_highlighted: false,
      position_number: '10034001',
      title: 'OMS (DCM)',
      is_overseas: true,
      create_date: '2006-09-20T00:00:00Z',
      update_date: '2017-06-08T00:00:00Z',
      effective_date: '2012-10-22T00:00:00Z',
      posted_date: '2012-10-22T00:00:00Z',
      status_code: null,
      status: null,
      description: {
        id: 90,
        last_editing_user: null,
        is_editable_by_user: false,
        date_created: '2018-03-08T18:00:33.241370Z',
        date_updated: '2018-03-13T15:55:00.018644Z',
        content: 'The Office Management Specialist (OMS) for the Deputy Chief of Mission (DCM) provides office management services in this Mission of six agencies, 387 American and Locally Employed Staff (LES) positions, and 279 guard force contractors.  The OMS manages Executive Office operations and positionstains the DCM’s calendar in support of overall Mission goals and objectives.  The OMS (DCM) drafts routine correspondence, memoranda and cables as appropriate and ensures documents are cleared and delivered in a timely fashion. The OMS serves as back up to the Ambassador’s OMS.  POCs areSandra McInturff, DCM OMS, McInturffSL@state.gov, 231-77-677-7000, ext. 7452; Samue Watson, DCM, WatsonSR@state.gov, 231-77-677-7000, ext. 7452. Please submit lobbying documents to the 360 Community Lobbying Center http://appcenter.state.sbu/CLC. (Revised 1/2017)\r\n\r\n',
        point_of_contact: null,
        website: null,
      },
      tour_of_duty: null,
      current_assignment: {
        user: 'John Doe',
        tour_of_duty: '2 YRS (2 R & R)',
        status: 'active',
        start_date: '2018-03-13T15:51:11.478084Z',
        estimated_end_date: '2020-03-13T15:51:11.478084Z',
      },
      post: {
        id: 160,
        code: 'LI6000000',
        location: {
          id: 103,
          country: 'Sierra Leone',
          code: '00A',
          city: 'Freetown',
          state: '',
        },
        tour_of_duty: '2 YRS (2 R & R)',
        cost_of_living_adjustment: 30,
        differential_rate: 30,
        danger_pay: 0,
        rest_relaxation_point: 'Paris',
        has_consumable_allowance: true,
        has_service_needs_differential: true,
        obc_id: null,
      },
      languages: [],
      bid_statistics: [
        {
          id: 6,
          bidcycle: 'Demo BidCycle 2018-03-13 15:51:10.980576+00:00',
          total_bids: 2,
          in_grade: 0,
          at_skill: 0,
          in_grade_at_skill: 0,
        },
      ],
      latest_bidcycle: {
        id: 1,
        name: 'Demo BidCycle 2018-03-13 15:51:10.980576+00:00',
        cycle_start_date: '2017-12-13T15:51:10.980554Z',
        cycle_deadline_date: '2018-04-13T15:51:10.980554Z',
        cycle_end_date: '2018-06-13T15:51:10.980554Z',
        active: true,
      },
    },
    {
      id: 60,
      grade: '04',
      skill: 'SECURITY (2501)',
      bureau: '(WHA) BUREAU OF WESTERN HEMISPHERIC AFFAIRS',
      organization: '(PORT AU PRIN) PORT AU PRINCE HAITI',
      classifications: [],
      representation: '[56082000] SPECIAL AGENT (Port-Au-Prince, Haiti)',
      is_highlighted: false,
      position_number: '56082000',
      title: 'SPECIAL AGENT',
      is_overseas: true,
      create_date: '2006-09-20T00:00:00Z',
      update_date: '2017-06-08T00:00:00Z',
      effective_date: '2016-10-02T00:00:00Z',
      posted_date: '2016-10-02T00:00:00Z',
      status_code: null,
      status: null,
      description: {
        id: 26,
        last_editing_user: null,
        is_editable_by_user: false,
        date_created: '2018-03-08T18:00:32.971745Z',
        date_updated: '2018-03-13T15:54:59.677323Z',
        content: '(updated August 2012) The incumbent serves as one of four Assistant Regional Security Officers (ARSO) in the Regional Security Office in a two year assignment with a rotating management portfolio that provides generous exposure to all aspects of Diplomatic Security’s overseas programs. RSO Port-au-Prince currently consists of a 6 member Marine Security Guard Detachment, 750+direct hire Mission guard force, 17 direct-hire Bodyguards, 2 FSN Investigators, 6 member Surveillance Detection Team, a 2-person Residential Security Unit, and one Office Management Specialist.  The incumbent has direct supervisory responsibilities.  The incumbent is expected to manage the Marine Security Guard program, Mission residential and physical security programs, operational and administrative oversight of the direct hire Mission guard force, and complex budget issues. The incumbent also coordinates security for large-scale Mission events and visiting high ranking USG officials and conducts personnel, terrorist, criminal, and counterintelligence investigations as required. Point of Contact is RSO Jeffrey M. Roberts.',
        point_of_contact: null,
        website: null,
      },
      tour_of_duty: null,
      current_assignment: {
        user: 'John Doe',
        tour_of_duty: '2 YRS (2 R & R)',
        status: 'active',
        start_date: '2018-03-13T15:51:13.126727Z',
        estimated_end_date: '2020-03-13T15:51:13.126727Z',
      },
      post: {
        id: 110,
        code: 'HA7000000',
        location: {
          id: 7,
          country: 'United States',
          code: '101',
          city: 'Chicago',
          state: 'IL',
        },
        tour_of_duty: '2 YRS (2 R & R)',
        cost_of_living_adjustment: 15,
        differential_rate: 20,
        danger_pay: 15,
        rest_relaxation_point: 'Miami',
        has_consumable_allowance: true,
        has_service_needs_differential: true,
        obc_id: null,
      },
      languages: [
        {
          id: 1,
          language: 'French (FR)',
          reading_proficiency: '2',
          spoken_proficiency: '2',
          representation: 'French (FR) 2/2',
        },
        {
          id: 20,
          language: 'Haitian Creole (HC)',
          reading_proficiency: '0',
          spoken_proficiency: '0',
          representation: 'Haitian Creole (HC) 0/0',
        },
      ],
      bid_statistics: [
        {
          id: 60,
          bidcycle: 'Demo BidCycle 2018-03-13 15:51:10.980576+00:00',
          total_bids: 0,
          in_grade: 0,
          at_skill: 0,
          in_grade_at_skill: 0,
        },
      ],
      latest_bidcycle: {
        id: 1,
        name: 'Demo BidCycle 2018-03-13 15:51:10.980576+00:00',
        cycle_start_date: '2017-12-13T15:51:10.980554Z',
        cycle_deadline_date: '2018-04-13T15:51:10.980554Z',
        cycle_end_date: '2018-06-13T15:51:10.980554Z',
        active: true,
      },
    },
  ],
};

export default resultsObject;
