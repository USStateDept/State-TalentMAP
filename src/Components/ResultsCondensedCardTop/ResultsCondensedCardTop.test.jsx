import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import ResultsCondensedCardTop from './ResultsCondensedCardTop';
import resultsObject from '../../__mocks__/resultsObject';
import bidListObject from '../../__mocks__/bidListObject';

describe('ResultsCondensedCardTopComponent', () => {
  const type = 'default';
  const favorites = [];
  it('is defined', () => {
    const wrapper = shallow(
      <ResultsCondensedCardTop
        position={resultsObject.results[0]}
        type={type}
        favorites={favorites}
        bidList={bidListObject.results}
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('is defined when isProjectedVacancy is true', () => {
    const wrapper = shallow(
      <ResultsCondensedCardTop
        position={resultsObject.results[0]}
        type={type}
        favorites={favorites}
        bidList={bidListObject.results}
        isProjectedVacancy
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('is defined when isRecentlyAvailable is true', () => {
    const wrapper = shallow(
      <ResultsCondensedCardTop
        position={resultsObject.results[0]}
        type={type}
        favorites={favorites}
        bidList={bidListObject.results}
        isRecentlyAvailable
      />,
    );
    expect(wrapper).toBeDefined();
  });

  it('matches snapshot', () => {
    const wrapper = shallow(
      <ResultsCondensedCardTop
        position={resultsObject.results[0]}
        type={type}
        favorites={favorites}
        bidList={bidListObject.results}
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it('matches snapshot when type is serviceNeed', () => {
    const wrapper = shallow(
      <ResultsCondensedCardTop
        position={resultsObject.results[0]}
        favorites={favorites}
        bidList={bidListObject.results}
        type="serviceNeed"
      />,
    );
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
});
