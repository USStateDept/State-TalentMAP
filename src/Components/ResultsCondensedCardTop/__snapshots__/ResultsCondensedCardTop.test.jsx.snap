// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsCondensedCardTopComponent matches snapshot 1`] = `
<div
  className="usa-grid-full condensed-card-top "
>
  <div
    className="usa-grid-full condensed-card-top-header-container"
  >
    <div
      className="usa-width-two-thirds condensed-card-top-header condensed-card-top-header-left"
    >
      <h3>
        OMS (DCM)
      </h3>
    </div>
    <div
      className="usa-width-one-third condensed-card-top-header condensed-card-top-header-right"
    >
      <dt>
        Grade:
      </dt>
       
      <dd>
        06
      </dd>
    </div>
  </div>
  <div
    className="usa-grid-full"
  >
    <Link
      replace={false}
      to="/details/10034001"
    >
      View position
    </Link>
  </div>
</div>
`;

exports[`ResultsCondensedCardTopComponent matches snapshot when type is serviceNeed 1`] = `
<div
  className="usa-grid-full condensed-card-top card-top-alternate"
>
  <div
    className="usa-grid-full condensed-card-top-header-container"
  >
    <div
      className="usa-width-two-thirds condensed-card-top-header condensed-card-top-header-left"
    >
      <span>
        <FontAwesome
          name="bolt"
        />
         
      </span>
      <h3>
        OMS (DCM)
      </h3>
    </div>
    <div
      className="usa-width-one-third condensed-card-top-header condensed-card-top-header-right"
    >
      <dt>
        Grade:
      </dt>
       
      <dd>
        06
      </dd>
    </div>
  </div>
  <div
    className="usa-grid-full"
  >
    <Link
      replace={false}
      to="/details/10034001"
    >
      View position
    </Link>
  </div>
</div>
`;
